src/nrtk_explorer/app/transforms.py

```diff
@@ -236,22 +233,18 @@ def compute_predictions_source_images(self, old_ids, ids):
         if len(ids) == 0:
             return
 
-        dataset = get_dataset(self.state.current_dataset)
-
         annotations = self.compute_annotations(ids)
         self.predictions_source_images = convert_from_predictions_to_first_arg(
             annotations,
-            dataset,
+            self.context.dataset,
             ids,
         )
 
-        # load ground truth annotations
-        dataset_annotations = dataset["annotations"]
         # collect annotations for each dataset_id
         annotations = {
             dataset_id: [
                 annotation
-                for annotation in dataset_annotations
+                for annotation in self.context.dataset.anns.values()
```

Comment:
If you are only using the values

```python
self.context.dataset.dataset['annotations']
```

avoids dependency on a preconstructed id-lookup index.